docs/template_plugin/tests/functional/op_reference/tan.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_core.hpp>
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+#include <tuple>
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+
+namespace {
+struct TanParams {
+    template <class IT>
+    TanParams(const ngraph::PartialShape& shape, const ngraph::element::Type& iType, const std::vector<IT>& iValues,
+              const std::vector<IT>& oValues)
+        :pshape(shape), inType(iType), outType(iType), inputData(CreateBlob(iType, iValues)), refData(CreateBlob(iType, oValues)) {}
+    ngraph::PartialShape pshape;
+    ngraph::element::Type inType;
+    ngraph::element::Type outType;
+    InferenceEngine::Blob::Ptr inputData;
+    InferenceEngine::Blob::Ptr refData;
+};
+
+class ReferenceTanLayerTest : public testing::TestWithParam<TanParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params.pshape, params.inType);
+        inputData = {params.inputData};
+        refOutData = {params.refData};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<TanParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "shape=" << param.pshape << "_";
+        result << "iType=" << param.inType << "_";
+        result << "oType=" << param.outType;
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const PartialShape& input_shape, const element::Type& input_type) {
+        const auto in = std::make_shared<op::Parameter>(input_type, input_shape);
+        const auto tan = std::make_shared<op::Tan>(in);
+        return std::make_shared<Function>(tan, ParameterVector {in});
+    }
+};
+
+TEST_P(ReferenceTanLayerTest, CompareWithHardcodedRefs) {
+    Exec();
+}
+
+std::vector<TanParams> generateTanCombinedParams() {
+    std::vector<TanParams> combinedParams {
+        TanParams(ngraph::PartialShape {5}, ngraph::element::i32, std::vector<int32_t> {-2, -1, 0, 1, 2},
+                  std::vector<int32_t> {2, -2, 0, 2, -2}),
+        TanParams(ngraph::PartialShape {5}, ngraph::element::i64, std::vector<int64_t> {-2, -1, 0, 1, 2},
+                  std::vector<int64_t> {2, -2, 0, 2, -2}),
+        TanParams(ngraph::PartialShape {5}, ngraph::element::u32, std::vector<uint32_t> {1, 2, 3, 4, 5},
```

Comment:
@ilyachur I don't know if need to support u32 u64, it's very strange to transform int to uint here

Reply:
@ilyachur  for example -2  to  0xFFFFFFFFFFFFFFFF - 1

Reply:
Why do you need negative numbers for uint data types?
Can you just change reference values?

Reply:
tan(2)=-2, but the output type is uint